boards: arm: mps2: Add initial FVP support for multiple mps2 boards

spchee opened the pull request:
- Added fvp support for the the following mps2 boards: an383, an385, an386 and an500. All common code from their respective devicetree files are now in `mps2_base.dtsi`
- Modified documentation to include all boards and include running with fvp.

pdgendt commented:
There are typos in the commit message 247e4041a6b677fe463f6e4cf83e52933d8afc31

`s/suport/support/` and `s/foar/for/`

spchee commented:
Have just fixed the typo, thanks for noticing.

nordicjm commented on boards/arm/mps2/board.cmake:
```diff
@@ -21,6 +21,10 @@ elseif(CONFIG_BOARD_MPS2_AN383)
+elseif(CONFIG_BOARD_MPS2_AN386)
+  set(SUPPORTED_EMU_PLATFORMS armfvp)
+  set(ARMFVP_BIN_NAME FVP_MPS2_Cortex-M4)
+
```
```suggestion
```

nordicjm commented on boards/arm/mps2/board.cmake:
```diff
@@ -21,6 +21,10 @@ elseif(CONFIG_BOARD_MPS2_AN383)
   -C armcortexm0plusct.VTOR=1
   )
 
```
```suggestion
```

nordicjm commented on boards/arm/mps2/doc/mps2_an386.rst:
```diff
@@ -0,0 +1,282 @@
+Overview
+********
+
+The mps2/an386 board configuration is used by Zephyr applications that run on
```
apply changes from previous commit

nordicjm commented on boards/arm/mps2/mps2_an385.dts:
```diff
@@ -11,6 +11,12 @@
 #include <zephyr/dt-bindings/input/input-event-codes.h>
 #include "mps2_base.dtsi"
 
+/* 	FVP does not support uart3 and uart4 whilst QEMU does.
+So if using QEMU, you might want to comment out the following two lines */
```
the indent here is very off and needs fixing

nordicjm commented on boards/arm/mps2/doc/mps2_v6m.rst:
```diff
@@ -1,13 +1,13 @@
 Overview
 ********
 
-The mps2/an383 board configuration is used by Zephyr applications that run on
```
why are all these being renamed? Think this breaks board listing @kartben ?

spchee committed:
soc: arm: mps2: Add support for mps2/an383 soc

Added new mps2 soc an383 to enable testing with ARM FVP.

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee committed:
boards: arm: mps2: Add support for mps2/an383

Added new mps2 board an383 to enable testing with ARM FVP.
Qualifier to build/run is mps2/an383

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee committed:
soc: arm: mps2: Add support for mps2/an386 soc

Added new mps2 soc an386 to enable testing with ARM FVP.

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee committed:
boards: arm: mps2: Add support for mps2/an386 in fvp

Added a support for new mps2 board an386 to enable testing with ARM FVP.
Qualifier to build/run application with board is mps2/an386.

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee committed:
boards: arm: mps2: Add support for mps2/an385 in fvp

Adds fvp support for an385.
Qualifier to build or run application with fvp is mps2/an385

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee committed:
soc: arm: mps2: Add support for mps2/an500 soc

Added new mps2 soc an383 to enable testing with ARM FVP.

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee committed:
boards: arm: mps2: Add support for mps2/an500 in FVP

Added support for an500 in fvp.
Qualifier to build and run is mps2/an500.

Signed-off-by: Samuel Chee <samche01@arm.com>

spchee commented:
@nordicjm Thanks for the review, have pushed an updated version which should resolve all your comments.

nordicjm commented on boards/arm/mps2/doc/mps2_an385.rst:
```diff
@@ -6,7 +6,7 @@ ARM V2M MPS2 AN385
 ********
 
-The mps2/an385 board configuration is used by Zephyr applications that run on
+The ``mps2/an385`` board configuration is used by Zephyr applications that run on
```
*board target

nordicjm commented on boards/arm/mps2/doc/mps2_an385.rst:
```diff
@@ -62,7 +62,7 @@ ARM V2M MPS2 provides the following hardware components:
 ==================
 
-The mps2/an385 board configuration supports the following hardware features:
+The ``mps2/an385`` board configuration supports the following hardware features:
```
*board target

nordicjm commented on boards/arm/mps2/doc/mps2_an386.rst:
```diff
@@ -0,0 +1,282 @@
+Overview
+********
+
+The ``mps2/an386`` board configuration is used by Zephyr applications that run on
```
*board target

nordicjm commented on boards/arm/mps2/doc/mps2_an386.rst:
```diff
@@ -0,0 +1,282 @@
+Supported Features
+==================
+
+The ``mps2/an386`` board configuration supports the following hardware features:
```
*board target

nordicjm commented on boards/arm/mps2/mps2_base.dtsi:
```diff
@@ -0,0 +1,268 @@
+			clocks = <&sysclk>;
+			current-speed = <115200>;
+		};
+		uart1: uart@40005000 {
```
missing newline

nordicjm commented on boards/arm/mps2/mps2_base.dtsi:
```diff
@@ -0,0 +1,268 @@
+		flash_sim0: flash_sim@0 {
+			compatible = "soc-nv-flash";
+			reg = <0x00000000 0x8000>;
+
```
```suggestion
```

nordicjm commented on boards/arm/mps2/mps2_base.dtsi:
```diff
@@ -0,0 +1,268 @@
+
+		gpio_button: mps2_fpgaio@40028008 {
+			compatible = "arm,mps2-fpgaio-gpio";
+
```
```suggestion
```

nordicjm commented on boards/arm/mps2/mps2_base.dtsi:
```diff
@@ -0,0 +1,268 @@
+
+		gpio_misc: mps2_fpgaio@4002804c {
+			compatible = "arm,mps2-fpgaio-gpio";
+
```
```suggestion
```

nordicjm commented on boards/arm/mps2/mps2_base.dtsi:
```diff
@@ -0,0 +1,268 @@
+
+	leds {
+		compatible = "gpio-leds";
+		led_0: led_0 {
+			gpios = <&gpio_led0 0>;
+			label = "USERLED0";
+		};
+		led_1: led_1 {
+			gpios = <&gpio_led0 1>;
+			label = "USERLED1";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button_0: button_0 {
+			label = "USERPB0";
+			gpios = <&gpio_button 0>;
+			zephyr,code = <INPUT_KEY_0>;
+		};
+		user_button_1: button_1 {
+			label = "USERPB1";
+			gpios = <&gpio_button 1>;
+			zephyr,code = <INPUT_KEY_1>;
+		};
```
spacing at odds with other part of file

spchee commented:
Thanks for the comments @nordicjm, most recent patch should have addressed everything

nordicjm commented on boards/arm/mps2/doc/mps2_armv6m.rst:
```diff
@@ -1,13 +1,13 @@
 
-The ``mps2/an383`` board target is used by Zephyr applications that run on
-the V2M MPS2 board. It provides support for the ARM Cortex-M0+ (AN383) CPU and
+Currently ``mps2/an383`` is the only mps2 armv6-m based board supported in Zephyr. The mps2/an383 board target is
```
```suggestion
Currently ``mps2/an383`` is the only mps2 armv6-m based board supported in Zephyr. The ``mps2/an383`` board target is
```

nordicjm commented on boards/arm/mps2/doc/mps2_armv7m.rst:
```diff
@@ -1,14 +1,15 @@
-The ``mps2/an385`` board target is used by Zephyr applications that run on
-the V2M MPS2 board. It provides support for the ARM Cortex-M3 (AN385) CPU and
-the following devices:
+The mps2 board target is used by Zephyr applications that run on
```
This is not correct, see https://docs.zephyrproject.org/latest/glossary.html#term-board-target for terms `mps2` is a board, `mps2/an385` is a board target, it would need updating throughout

nordicjm reviewed (commented):
minor nit. in doc

spchee commented:
Have addressed your nits :) @nordicjm
